typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,89 +1,727 @@
+    return module;
+  }
+
+  protected async updateRoutingIsm({
+    delta,
+    config,
+    logger,
+  }: {
+    delta: RoutingIsmDelta;
+    config: RoutingIsmConfig;
+    logger: Logger;
+  }): Promise<AnnotatedEthersV5Transaction[]> {
+    const deployedIsmAddress = this.args.addresses.deployedIsm;
+    const routingIsmInterface = DomainRoutingIsm__factory.createInterface();
+
+    const updateTxs = [];
+
+    // Enroll domains
+    for (const originDomain of delta.domainsToEnroll) {
+      // get name of origin chain
+      const origin = this.multiProvider.getChainName(originDomain);
+      logger.debug(
+        `Reconfiguring preexisting routing ISM for origin ${origin}...`,
+      );
+      const ism = await this.deploy({
+        config: config.domains[origin],
+      });
+
+      const tx = createAnnotatedEthersV5Transaction({
+        annotation: `Setting new ISM for origin ${origin}...`,
+        chainId: this.domainId,
+        to: deployedIsmAddress,
+        data: routingIsmInterface.encodeFunctionData('set(uint32,address)', [
+          originDomain,
+          ism.address,
+        ]),
+      });
+
+      updateTxs.push(tx);
+    }
+
+    // Unenroll domains
+    for (const originDomain of delta.domainsToUnenroll) {
+      const tx = createAnnotatedEthersV5Transaction({
+        annotation: `Unenrolling originDomain ${originDomain} from preexisting routing ISM at ${deployedIsmAddress}...`,
+        chainId: this.domainId,
+        to: deployedIsmAddress,
+        data: routingIsmInterface.encodeFunctionData('remove(uint32)', [
+          originDomain,
+        ]),
+      });
+      updateTxs.push(tx);
+    }
+
+    // Transfer ownership if needed
+    if (delta.owner) {
+      const tx = createAnnotatedEthersV5Transaction({
+        annotation: `Transferring ownership of routing ISM...`,
+        chainId: this.domainId,
+        to: deployedIsmAddress,
+        data: routingIsmInterface.encodeFunctionData(
+          'transferOwnership(address)',
+          [delta.owner],
+        ),
+      });
+      updateTxs.push(tx);
+    }
+
+    return updateTxs;
+  }
+
+  protected updateFallbackRoutingIsm({
+    owner,
+    domainIds,
+    submoduleAddresses,
+  }: {
+    owner: Address;
+    domainIds: Domain[];
+    submoduleAddresses: Address[];
+  }): AnnotatedEthersV5Transaction[] {
+    const routingIsmInterface =
+      DefaultFallbackRoutingIsm__factory.createInterface();
+
+    const updateTx = createAnnotatedEthersV5Transaction({
+      annotation: 'Updating fallback routing ISM ...',
+      chainId: this.multiProvider.getDomainId(this.args.chain),
+      to: this.args.addresses.deployedIsm,
+      data: routingIsmInterface.encodeFunctionData(
+        'initialize(address,uint32[],address[])',
+        [owner, domainIds, submoduleAddresses],
+      ),
+    });
+    return [updateTx];
+  }
+
+  protected async deployRoutingIsmSubmodules({
+    config,
+  }: {
+    config: RoutingIsmConfig;
+  }): Promise<Address[]> {
```

Comment:
why wouldnt this return an object? similar to the `WithAddress<RoutingIsmConfig>` type

Reply:
you mean `WithAddress<RoutingIsmConfig>[]` instead?